src/jit/codegenxarch.cpp

```diff
@@ -6674,8 +6670,6 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
     }
     else // non-overflow checking cast
     {
-        noway_assert(size < EA_PTRSIZE || srcType == dstType);
```

Comment:
We delete this assert for both platforms. The logic was not changed, I replaced size variable with exact using srcSize or dstSize, because it was hard to prove that this assert could be deleted.
So there are no any superpmi asm diffs for x64.